frontend/src/components/TableTaskRow.tsx

```diff
@@ -33,6 +41,63 @@ export const TableTaskRow: React.FC<TableTaskRowProps> = ({ task }) => {
     userInfoSelector,
     shallowEqual,
   );
+  const allUsers = useSelector<RootState, RoadmapUser[] | undefined>(
+    roadmapUsersSelector,
+    shallowEqual,
+  );
+  const allCustomers = useSelector<RootState, Customer[] | undefined>(
+    allCustomersSelector,
+    shallowEqual,
+  );
+  const [missingRatings, setMissingRatings] = useState<Customer[] | undefined>(
+    undefined,
+  );
+  const [missingDevRatings, setMissingDevRatings] = useState<
+    RoadmapUser[] | undefined
+  >([]);
+  const [userRatingMissing, setUserRatingMissing] = useState<Boolean>(true);
+
+  /*
+    AdminUsers can see missing customer and developer ratings
+    DeveloperUser can see missing developer ratings
+    CustomerUser and BusinessUser can see their own missing ratings
+  */
+  useEffect(() => {
+    if (userInfo?.type === UserType.AdminUser) {
+      const givenRatings = task.ratings
+        .map((rating) => {
+          return rating.forCustomer;
+        })
+        .filter((value) => value !== null);
+
+      const unratedCustomers = allCustomers?.filter(
+        (customer) => !givenRatings.includes(customer.id),
+      );
+      setMissingRatings(unratedCustomers);
+    }
+
+    if (
+      userInfo?.type === UserType.AdminUser ||
+      userInfo?.type === UserType.DeveloperUser
+    ) {
+      const ratingIds = task.ratings.map((rating) => rating.createdByUser);
+      const developers = allUsers?.filter((user) => user.type === 279715);
+      const missingDevs = developers?.filter(
+        (developer) => !ratingIds.includes(developer.id),
+      );
+      setMissingDevRatings(missingDevs);
+    }
+
+    if (
+      userInfo?.type === UserType.CustomerUser ||
+      userInfo?.type === UserType.BusinessUser
+    ) {
+      // if task doesn't have ratings from the user that is logged in, display icon to them.
+      setUserRatingMissing(
+        !task.ratings.find((rating) => rating.createdByUser === userInfo?.id),
```

Comment:
```suggestion
        !task.ratings.some((rating) => rating.createdByUser === userInfo?.id),
```

`[].some` on predikaatti versio `[].find`:lle, joten semanttisesti tähän sopivampi, joskin molemmat toimivat